### DISCUSSION TITLE
Problems with my LibreVNA

### DISCUSSION THREAD

**Discussion by digidiver11** [jankae/LibreVNA#237] [General]
Hello,

I bought the LibreVNA some time ago.
Unfortunately, I never used it properly because I often used the VNWA from DG8SAQ.

Now I have newer projects with much higher frequencies.
I have always kept the LibreVNA up to date with firmware and the GUI.
The latest update V1.5.0 is also fully installed.

For some time now I have noticed that port 2 does not seem to work properly on my LibreVNA. But perhaps I have also made a mistake.
At the beginning three screenshots, which were performed after a SOLT for both ports.

I would be very pleased if you could give me some tips on this behavior of the LibreVNA, possibly also on how to repair it.
One more note, the device has not yet been used. It can therefore not have been damaged by excessive input levels or voltages.

Thank you in advance for your feedback.

I wish you all a happy, healthy and peaceful New Year.

D. Kochan    from Berlin

![All_Ports_Open](https://github.com/jankae/LibreVNA/assets/25418080/e3415907-4397-4f40-bdbe-1849f585a2c1)
![Port_1-close__Port_2-open](https://github.com/jankae/LibreVNA/assets/25418080/c08658ff-6b89-4e8f-927d-6d120c410c06)
![Port_1-open__Port_2-close](https://github.com/jankae/LibreVNA/assets/25418080/9ff65890-5673-4256-a7c1-64bd489d5a1f)
![All_Ports__LOAD](https://github.com/jankae/LibreVNA/assets/25418080/3fcc6d67-9fbd-4516-aba2-d505b7fb6a65)

**Comment by jankae** [comment]
This very much looks like a SOL(T) calibration on port 1 only. The enabled calibration type is not visible in your screenshots (should be to the right of your toolbar). Please save your calibration (Menu Calibration->Save) and attach that file. If my suspicion is true, this should be very easy to see in the file.

**Comment by digidiver11** [comment]
Addendum to the images, sorry !

Picture1:   All_Ports_Open
Picture2:   Port_1-close__Port_2-open
Picture3:   Port_1-open__Port_2-close
Picture3:   All Ports with 50 Ohm Loads

Sorry !

**Comment by digidiver11** [comment]
Hello jankee,

thanks for the quick reply
Enclosed the requested date

Thanks
[SOLT_1_1_00M-6_00G_501pt.cal___.txt](https://github.com/jankae/LibreVNA/files/13843900/SOLT_1_1_00M-6_00G_501pt.cal___.txt)

**Comment by jankae** [comment]
Thanks, this calibration is indeed only valid for port 1. I guess you tried to set a SOL calibration on port 1 and a SOL calibration on port 2. This is not supported at the same time. The math for such an operation is possible but the software does not support this (because it would have to be treated as two active calibrations at the same time and there is only one calibration object).

If you want to have a valid calibration on both ports, you need to add a through measurement and enable the SOLT calibration for both ports:

Building on top of your already existing calibration:
1. Add a through measurement:
![image](https://github.com/jankae/LibreVNA/assets/16818866/6ba59499-aded-4ed2-a970-413fa26e6826)

2. Measure the through standard:
![image](https://github.com/jankae/LibreVNA/assets/16818866/ccc51146-c8f5-4609-831d-06f8e4bc17bc)

3. Enable the SOLT calibration for both ports:
![image](https://github.com/jankae/LibreVNA/assets/16818866/daa5a5ac-51ab-43b1-b2a7-a84d865b676e)

**Reply by digidiver11** [reply]
Thank you for your quick reply.

I will try it out tomorrow and get back to you with the result.
Thank you very much.

Is there a PDF file for your software ???
I had another small, stupid problem.

Due to a careless click, the left guide with the "Traces" disappeared. I was not able to bring them back to the left side. So I closed the software and searched for a config file. Since I couldn't find one, I checked the registry and simply deleted the complete registry entry under user/software/libreVNA.

After that I had the old state again.
This is definitely not the right way!

Thank you very much, also for this great VNA !!!!


D. Kochan

**Reply by jankae** [reply]
> Is there a PDF file for your software ???

There is the user manual but it doesn't cover every option: https://github.com/jankae/LibreVNA/blob/master/Documentation/UserManual/manual.pdf

> Due to a careless click, the left guide with the "Traces" disappeared. I was not able to bring them back to the left side.

Check out the menu: Window->Docks->Traces

**Reply by digidiver11** [reply]
Thanks for answering, but this dock was then located crosswise under the top menu bar. I already knew this way "Window->Docks->Traces" !

But I was not able to move this dock to the left side, not even by dragging it with the mouse ...... . That was my problem ;-)).

**Comment by digidiver11** [comment]
Hello,

unfortunately I only got around to implementing your tips on calibration today.

I did the SOL calibrations for both ports and then did the T-calibration in both directions and saved the whole thing as a calibration.
Unfortunately, there is still a clearly visible error on port 2. Here is a screenshot.

It gets really bad when port 2 is open or closed.
Then only garbage is displayed on port 2.

I have the feeling that either I don't understand the calibration correctly or the device is really defective.

I don't know what to do anymore.

With kind regards
![Port2___failed_calibration](https://github.com/jankae/LibreVNA/assets/25418080/dc5b1452-453c-4eb3-9d77-1f671f3057cc)
[Cal-Datas.zip](https://github.com/jankae/LibreVNA/files/13928624/Cal-Datas.zip)

**Comment by jankae** [comment]
Hi,
you calibration data now includes the through measurement (by the way: a single through measurement is enough, no need to do that for both directions - but the additional measurement also doesn't hurt, it is simply not used by the calibration).

You seem to have saved the file while the calibration was disabled ("type": "None" near the end of the file). In your screenshot the calibration is enabled but I can't see which type you have enabled (should be "SOLT_12"). 

Please try the following:
1. Start the GUI, load your calibration file (through the menu with Calibration->Load)
2. Go to the calibration measurements (Calibration->Calibration Measurements)
3. Enable the SOLT_12 calibration by double-clicking this item:
![image](https://github.com/jankae/LibreVNA/assets/16818866/c025ba97-26b4-4d30-b560-262193b273b4)
4. Verify that the fields on the bottom right show these values:
![image](https://github.com/jankae/LibreVNA/assets/16818866/c24b1176-46fc-4392-8e4e-564d3ec0567b)
5. Connect the load standard to port 2. It should show up as a dot in the center of the smith chart (like in your screenshots for port 1)

If this doesn't work, please double-check your setup and especially look out for differences to when you took the calibration measurements. I don't really believe in a hardware problem because your raw measurements (included in the calibration files) look very reasonable with not much of a difference between port 1 and 2.

**Reply by digidiver11** [reply]
Hello,

Thank you for your quick reply.

I think I have now understood (and documented !!!) where my mistake was.

After the calibration data was loaded, the calibration " Solt 1,2 had to be ACTIVATED under "Calibration-Measurements". That was the problem.

Here is a screenshot.

Thank you very much !!!
![Korrekte_Anwendung__SOLT1+2](https://github.com/jankae/LibreVNA/assets/25418080/8bb4883c-8b70-41fe-9eb1-2bf4723c7a24)

**Reply by jankae** [reply]
Great, thanks for confirming that it works now :) I will add this to my todo list and show a detailed example on how to create a standard SOLT on both ports (probably by far the most used calibration type) in the user manual soon

**Reply by digidiver11** [reply]
Hello jankae,

Once I understood the principle, it's not that difficult to get started. But I would really appreciate it if you could show the various calibration methods in your manual. But I understand that it all takes a lot of work and time. But I would like to thank you again for this great VNA, which would not really be possible in hobby circles.

Allow me to ask you a very small question:

The device gets quite warm during operation. Does it make sense to fit the device with external heat sinks or is there another way to optimize it?

With best regards

**Reply by jankae** [reply]
> The device gets quite warm during operation. Does it make sense to fit the device with external heat sinks or is there another way to optimize it?

The heat will not damage the device (it is still far below the limit for the components). But it may bother you when you handle it and it will also increase thermal drift.

You can add an additional heat sink (and also a fan if you want) but you will have to come up with that yourself. This is something that a couple of other users have already done, you will likely find images of their constructions if you search for it.
